store/sqlstore/session_store.go

```diff
@@ -52,19 +52,16 @@ func (me SqlSessionStore) Save(session *model.Session) (*model.Session, *model.A
 
 	session.PreSave()
 
-	tcs := me.Team().GetTeamsForUser(session.UserId)
+	tempMembers, err := me.Team().GetTeamsForUser(session.UserId)
 
 	if err := me.GetMaster().Insert(session); err != nil {
 		return nil, model.NewAppError("SqlSessionStore.Save", "store.sql_session.save.app_error", nil, "id="+session.Id+", "+err.Error(), http.StatusInternalServerError)
 	}
 
-	rtcs := <-tcs
-
-	if rtcs.Err != nil {
-		return nil, model.NewAppError("SqlSessionStore.Save", "store.sql_session.save.app_error", nil, "id="+session.Id+", "+rtcs.Err.Error(), http.StatusInternalServerError)
+	if err != nil {
```

Comment:
hmm looks like err for SqlSessionStore.Save is being checked twice here. Should one of it be for GetTeamsForUser call?

Reply:
You are right, I'll fix this

Reply:
I have a doubt, in the old code the informed error is **where** the error happens, not the cause.

![image](https://user-images.githubusercontent.com/1508272/59384662-b1a77880-8d30-11e9-8c28-c844526f9ebe.png)

Is this ok ?

Reply:
I'm not entirely sure if I understood that, but looking at the previous code I'm not sure if it was intended this way? @jespino can you review this when you get a chance please?

Reply:
Sure, I'll take a look in this PR tomorrow during my morning, I live in Spain, so it's a bit late here. Thanks @mounicapaladugu

Reply:
@rvillablanca I don't really understand the question about the error, can you explain it a bit more?

Reply:
Ok, now I understand the question.

The error is ok. It is happening in the Save method, another thing is this methods relays in other method to do this. Anyway we want to rethink or the errors part.

Reply:
Sorry as you have seen my english is not the best

Reply:
@rvillablanca Don't worry, I'm not a native speaker either. I guess you are a Spanish native speaker like me, so... if you have questions and you prefer to ask me in Spanish, you can contact me directly in https://community.mattermost.com :)